scripts/mvf_rechunk.py

```diff
@@ -0,0 +1,231 @@
+                for array_name in src_ts['chunk_info']:
+                    if array_name == 'flags' and src_stream != stream_name:
+                        # Upgraded flags completely replace the source stream's
+                        # flags, rather than augmenting them. Thus, data lost in
+                        # the source stream has no effect.
+                        continue
+                    lost_flags = arrays[(src_stream, array_name)].lost_flags
+                    lost_flags = lost_flags.rechunk(flags_array.data.chunks[:lost_flags.ndim])
+                    # weights_channel doesn't have a baseline axis
+                    while lost_flags.ndim < flags_array.data.ndim:
+                        lost_flags = lost_flags[..., np.newaxis]
+                    lost_flags = da.broadcast_to(lost_flags, flags_array.data.shape,
+                                                 chunks=flags_array.data.chunks)
+                    flags_array.data |= lost_flags
+
+    # Apply the rechunking specs
+    for spec in args.spec:
+        key = (spec.stream, spec.array)
+        if key not in arrays:
+            raise RuntimeError('{}/{} is not a known array'.format(spec.stream, spec.array))
+        arrays[key].data = arrays[key].data.rechunk({0: spec.time, 1: spec.freq})
+
+    # Write out the new data
+    dest_store = NpyFileChunkStore(args.dest)
+    stores = []
+    for array in arrays.values():
+        full_name = dest_store.join(array.chunk_info['prefix'], array.array_name)
+        dest_store.create_array(full_name)
+        stores.append(dest_store.put_dask_array(full_name, array.data))
+        array.chunk_info['chunks'] = array.data.chunks
+    stores = da.compute(*stores)
+    # put_dask_array returns an array with an exception object per chunk
+    for result_set in stores:
+        for result in result_set.flat:
+            if result is not None:
+                raise result
+
+    # Fix up chunk_info for new chunking
+    for stream_name in streams:
+        sts = view_capture_stream(telstate, cbid, stream_name)
+        chunk_info = sts['chunk_info']
+        for array_name in chunk_info.keys():
+            chunk_info[array_name] = arrays[(stream_name, array_name)].chunk_info
+        sts.wrapped.delete('chunk_info')
+        sts.wrapped['chunk_info'] = chunk_info
+        # s3_endpoint_url is for the old version of the data
+        sts.wrapped.delete('s3_endpoint_url')
+        if args.s3_endpoint_url is not None:
+            sts.wrapped['s3_endpoint_url'] = args.s3_endpoint_url
+
+    # Write updated RDB file
+    url_parts = urllib.parse.urlparse(args.source, scheme='file')
+    dest_file = os.path.join(args.dest, args.new_prefix or cbid, os.path.basename(url_parts.path))
+    os.makedirs(os.path.dirname(dest_file), exist_ok=True)
+    writer = RDBWriter(client=telstate.backend)
+    writer.save(dest_file)
+
+
+if __name__ == '__main__':
+    try:
+        main()
```

Comment:
I've always done it because [Guido does](https://www.artima.com/weblogs/viewpost.jsp?thread=4829). Apart from the advice there, I found another advantage on StackOverflow: it keeps the variables used by the main code local to `main`, so that they don't accidentally leak into other functions.

Reply:
Ah, so Guido talks about "a main() function that's easy to invoke in other contexts, e.g. from the interactive Python prompt when you feel like experimenting." We don't really import `main` elsewhere.

And the localised variables are exactly why I *don't* like `main()` 😀 Instead of importing `main()` in IPython, I prefer to `%run` the script and then inspect the local namespace after the script is done for debugging and enlightenment. Since we don't import scripts, for me the risk of leakage is low while the convenience of non-main is high...

Just an opinion, happy to keep this as is.